### ISSUE TITLE
Getting "Each child in a list should have a unique "key" prop." warning when using createIcon with multiple paths

### ISSUE THREAD

**Issue by oahmaro** [chakra-ui/chakra-ui#5127]
### Description

When i use createIcon function with multiple paths i get "Each child in a list should have a unique "key" prop." warning

### Link to Reproduction

https://codesandbox.io/s/exciting-drake-ch4k0?file=/src/index.tsx

### Chakra UI Version

1.6.12

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by zgid123** [comment]
I encountered this issue too. It seems we need to assign the key for each path if the prop path is array by using `React.cloneElement` or else we need to assign key for each path manually. I will open a PR

**Comment by segunadebayo** [comment]
Hi @oahmaro,

Based on the TS types, I can see why you'd want to pass an array for the paths. A workaround will be to wrap the paths in a fragment.

```jsx
const Apple = createIcon({
  displayName: "Apple",
  path: (
    <>
      <path d="..." />
      <path d="..." />
    </>
  ),
  viewBox: "0 0 512 512"
});
```

**Comment by oahmaro** [comment]
@segunadebayo Types and also i found this comment on the docs that made me think that i might need to pass an array.


<img width="754" alt="Screen Shot 2021-11-22 at 15 46 54" src="https://user-images.githubusercontent.com/15835932/142872840-635bd88e-e746-45cd-804c-090eebf2c1e4.png">

**Comment by anubra266** [comment]
@oahmaro he's only suggesting it as a workaround, a fix is already being worked on. The PR is linked to this issue.

**Comment by oahmaro** [comment]
@anubra266 Thank you

**Comment by Philzen** [comment]
Just discovered that the solution implemented via #5129 currently breaks in React 19 – for reasons i've yet to discover.

However i was able to come up with an alternative implementation inspired by [this post](https://www.devgem.io/posts/react-solving-the-key-prop-for-each-child-error-in-map-method-a-guide) that does not trigger the warning:

```tsx
import { Fragment } from 'react'

// …

export function createIcon(options: CreateIconOptions) {
  const { viewBox = '0 0 24 24', d, displayName, defaultProps = {} } = options
  const path = Array.isArray(options.path)
    ? options.path.map((el, idx) => (
        <Fragment key={el.key || idx}>{el}</Fragment>
      ))
    : options.path

  const Comp = (props) => (
    <Icon ref={props.ref} viewBox={viewBox} {...defaultProps} {...props}>
      {path || <path fill="currentColor" d={d} />}
    </Icon>
  )

  Comp.displayName = displayName

  return Comp
}
```

May be worth monitoring if that issue persists, and if it does, probably updating the current `createIcon()` implementation.
